### PR TITLE
Efx environment

### PR DESCRIPTION
**Pull request by johncurley**
EFX environment code port. There is currently an error flush routine for OpenAL as else the sound cuts out. I'll attempt to find that error and clean the code up properly at some stage but it would be good to test this on other operating systems as EFX is software accelerated. Seems to function well on pulse audio.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -27,3 +27,6 @@ CMakeFiles/
 
 # exclude doxygen output
 doc/doxygen/html/*
+
+# exclude project files
+*.kdev4
```
Comment (Xottab-DUTY) [inline .gitignore]: This should go as another pull request

**Comment by Xottab-DUTY** [comment]
Please, don't remove EAX code

**Comment by johncurley** [comment]
I'll rewrite the environment code to have parallel code paths for both EFX and EAX support with the old code.

**Comment by johncurley** [comment]
I'll see if I can find the source of the error causing the Linux sound to stop using EFX.

**Comment by johncurley** [comment]
I'm going to fix all this up and repull.
